src/app/auth/services/auth.service.ts

```diff
@@ -10,14 +14,61 @@ const INITIAL_STATE: AuthState = {
 @Injectable({ providedIn: 'root' })
 export class AuthService {
   private readonly _authState = signal(INITIAL_STATE);
+  private http = inject(HttpClient);
 
   user = computed(() => this._authState().user);
   status = computed(() => this._authState().status);
+  isLoading = signal(false);
+
+  saveUserToLocalStorage(userData: SignupRequest): void {
+    localStorage.setItem('user', JSON.stringify(userData));
+  }
+
+  getUserFromLocalStorage(): SignupRequest | null {
+    const data = localStorage.getItem('user');
+    localStorage.removeItem('user');
+    if (!data) return null;
+
+    try {
+      return JSON.parse(data) as SignupRequest;
+    } catch {
+      return null;
+    }
+  }
 
   logout() {
     this._authState.set({
       user: null,
       status: AuthStatus.UNAUTHENTICATED,
     });
   }
+
+  signup(signupRequest: SignupRequest): Observable<{ success: boolean; status: number }> {
+    if (this.isLoading()) {
+      return EMPTY;
+    }
+
+    this.isLoading.set(true);
+    this.saveUserToLocalStorage(signupRequest);
+
+    return this.http.post<{ success: boolean; status: number }>(`${environment.API_URL}/auth/signup`, {
+      dni: signupRequest.dni,
+      firstName: signupRequest.firstName,
+      lastName: signupRequest.lastName,
+      email: signupRequest.email,
+      password: signupRequest.password,
+      address: signupRequest.address,
+      phoneNumber: signupRequest.phoneNumber,
+      cuit: signupRequest.cuit,
+      birthDate: signupRequest.birthDate
+    }, { responseType: 'text' as 'json' }).pipe(
+      tap({
+        next: () => this.isLoading.set(false),
+        error: () => this.isLoading.set(false),
+        complete: () => this.isLoading.set(false)
+      })
```

Comment:
Creo que podemos onelinear esto

```suggestion
      tap(finalize(() => this.isLoading.set(false)))
```